container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/user_services_functions/start_user_services.go

```diff
@@ -129,11 +134,20 @@ func StartUserServices(
 		return successfulServicesPool, failedServicesPool, nil
 	}
 
+	for _, serviceRegistrationToStart := range serviceRegistrationsToStart {
```

Comment:
Similar to the service registrations map at the service network layer, we are using the one at the Docker backend layer to keep track of the service status and know if a call to `StartRegisteredUserServices` is the first start after registration or a restart.